crates/net/discv4/src/test_utils.rs

```diff
@@ -265,23 +265,23 @@ pub fn rng_endpoint(rng: &mut impl Rng) -> NodeEndpoint {
 /// Generates a random [`NodeRecord`] using the provided random number generator.
 pub fn rng_record(rng: &mut impl RngCore) -> NodeRecord {
     let NodeEndpoint { address, udp_port, tcp_port } = rng_endpoint(rng);
-    NodeRecord { address, tcp_port, udp_port, id: rng.gen() }
+    NodeRecord { address, tcp_port, udp_port, id: B512::random() }
```

Comment:
These stop using `rng` which is not entirely correct as discussed.

Undoing these changes when rand_08 is removed will be a nightmare; should at least leave a `// TODO(rand)` for each `::random()`?